engine/apps/schedules/ical_utils.py

```diff
@@ -90,14 +93,9 @@ def users_in_ical(
         (Q(username__in=usernames_from_ical) | Q(email__lower__in=emails_from_ical))
     ).distinct()
 
-    if organization.is_rbac_permissions_enabled:
-        # it is more efficient to check permissions on the subset of users filtered above
-        # than performing a regex query for the required permission
-        users_found_in_ical = [u for u in users_found_in_ical if {"action": required_permission.value} in u.permissions]
-    else:
-        users_found_in_ical = users_found_in_ical.filter(role__lte=required_permission.fallback_role.value)
-
-    return list(users_found_in_ical)
```

Comment:
this makes things much cleaner. `users_in_ical` shouldn't need to know anything about how to conditionally check permissions (ie. `is_rbac_permissions_enabled`) or the structure of RBAC permissions (`{"action": required_permission.value}`).

See the new [`LegacyAccessControlCompatiblePermission.user_has_permission` method](https://github.com/grafana/oncall/pull/4784/files#diff-83058af8678d629be3d878a5947537c15c01abf6ef2eb7ba31d08b8febda0853R100-R101)